### ISSUE TITLE
[agw] replacing std::unordered_map with google::protobuf::Map in map.h

### ISSUE THREAD

**Issue by ganeshg87** [magma/magma#10869]
### Description 

- `magma::map_s` interface is currently being used on the AMF task to allocate both task (UE ids hashtables) and per-UE state. 
- `std::unordered_map` is used under the hood for this interface wrapper. 

### Goal
- Protobuf data structures and objects should be the standard to allocate state across the MME codebase, the standard map can be updated to use Protobuf's map interface.
- Blocked by #11189 due to GUTI key lookup.

**Comment by shahrahman-fb** [comment]
Closing this issue as part of a repository cleanup effort. This issue has had no activity for over 2 years.

If this issue is still relevant, please reopen it or file a new issue with updated information. We are actively maintaining the project again and want to focus on current, actionable issues.

See: #15833 (Phase 1 cleanup), #15828 (tracking issue)
